### PR TITLE
feat: new sheet component

### PR DESCRIPTION
**Pull request by muselesscreator**
Implements a basic "Sheet" component, which serves as a basic UI container element bound to and spanning one edge of the window.
Supports light/dark theming, as well as content-blocking.

https://www.figma.com/file/eGmDp94FlqEr4iSqy1Uc1K/Paragon-2021?node-id=7520%3A16719

### CODE REVIEW

File: src/Sheet/SheetContainer.test.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    expect(rootEl.classList.contains('sheet-container')).toEqual(true);
+    const childEl = rootEl.querySelector(`#${childId1}`);
+    expect(childEl.innerHTML).toEqual(childContent1);
+  });
+
+  it('only adds itself once, when invoked multiple times', () => {
+    mount(<SheetContainer>{child1}</SheetContainer>);
+    mount(<SheetContainer>{child2}</SheetContainer>);
+    const rootEls = document.querySelectorAll('#sheet-root');
+    expect(rootEls.length).toBeGreaterThan(0);
```
Comment (abutterworth) [inline src/Sheet/SheetContainer.test.jsx:27]: Should this be `1`?

File: src/Sheet/Sheet.scss [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.pgn__sheet-skrim {
+  width: 100%;
+  height: 100%;
+  background-color: rgba(196,196,196,0.5);
+  position: fixed;
```
Comment (abutterworth) [inline src/Sheet/Sheet.scss:5]: We should probably specify a z-index. Here is where z-indexes are currently set: https://github.com/edx/paragon/blob/80ce916912cc13bafe81d2e90acb97b582cbb65f/scss/core/_variables.scss#L610-L611

We might want to add:
```
$zindex-sheet-backdrop:             1031 !default;
$zindex-sheet:                      1032 !default;
```

File: src/Sheet/index.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          role="presentation"
+        />
+        <div
+          className={classNames(
+            'pgn__sheet-component',
+            `pgn__sheet__${variant}`,
+            position,
+
+          )}
+          role="alert"
```
Comment (abutterworth) [inline src/Sheet/index.jsx:57]: @wittjeff I know there was some discussion on the sheet component using the alert role, but I'm guess this role would be odd if the sheet were used for a side nav or some use case like it. Is this something we should bake in, make configurable, or leave to child elements to define?

File: src/Sheet/SheetContainer.test.jsx (hunk 1 above)
Comment (muselesscreator) [reply src/Sheet/SheetContainer.test.jsx:27]: yeah, but should also have been `toEqual`

File: src/Sheet/Sheet.scss (hunk 2 above)
Comment (muselesscreator) [reply src/Sheet/Sheet.scss:5]: reasonable.

File: src/Sheet/index.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+class Sheet extends React.Component {
+  constructor(props) {
+    super(props);
+    this.blockClicks = this.blockClicks.bind(this);
+  }
+
+  blockClicks(event) {
+    event.stopPropagation();
+  }
```
Comment (abutterworth) [inline src/Sheet/index.jsx:27]: The sheet doesn't currently lock focus inside the sheet if it's set to blocking. I suggest using this library: https://github.com/theKashey/react-focus-on to do this work. It does a great job of handling scrolling, clicking, and focus blocking very well.

**Review by abutterworth — CHANGES_REQUESTED** [review]
Looks great so far. A couple things on z-index and focus locking above

File: src/Sheet/Sheet.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  top: 0px;
+  left: 0px;
+  font-size: 18px;
```
Comment (abutterworth) [inline src/Sheet/Sheet.scss:8]:
```suggestion
```

File: src/Sheet/Sheet.scss [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  &.hidden {
+    display: none;
+  }
+  z-index: $zindex-sheet-backdrop;
+}
+.pgn__sheet-component {
+  position: fixed;
+  padding: 20px;
```
Comment (abutterworth) [inline src/Sheet/Sheet.scss:16]:
```suggestion
  padding: 1.25rem;
```

File: src/Sheet/Sheet.scss [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  background-color: white;
+  z-index: $zindex-sheet;
+  &.pgn__sheet__dark {
+    background-color: rgb(0,38,43);
+    color: rgb(242,240,239);
```
Comment (abutterworth) [inline src/Sheet/Sheet.scss:21]:
```suggestion
    background-color: $primary-500;
    color: $light-300;
```

File: src/Sheet/index.jsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        className={classNames(
+          'pgn__sheet-component',
+          `pgn__sheet__${variant}`,
+          position,
+        )}
+        role="alert"
+        {...{
+          'aria-live': 'polite',
+          'aria-atomic': 'true',
+        }}
```
Comment (abutterworth) [inline src/Sheet/index.jsx:39]:
```suggestion
        aria-live="polite"
        aria-atomic="true"
```
You shouldn't need to do this, the linter should accept aria attributes on jsx

**Review by abutterworth — APPROVED** [review]
One minor thing about syntax for getting aria attributes onto the jsx component that's minor.  Let's follow up with Jeff on the aria-role when he can see an example.

**Pull request merged**